app/serializers/order_serializer.rb

```diff
@@ -62,13 +68,44 @@ def ordered_by_current_user
     object.user == current_user
   end
 
+  def current_user_debt_permitted
+    current_user.total_debt.to_f > Dish::MAX_DEBT
+  end
+
   def from_today
     object.from_today?
   end
 
+  def order_owner
+    object.user.name
+  end
+
+  def current_user_ordered_dishes
+    return "You didn't order any dish." if current_user_dishes.blank?
+    current_user_dishes.map(&:name).join(', ')
+  end
+
+  def current_user_dishes_price
+    return '0.00' if current_user_dishes.blank?
+    current_user_dishes
+      .pluck('SUM(price_cents/user_dishes_count)')[0].to_s.insert(-3, '.')
+  end
+
+  def current_user_shipping_cost
+    return object.shipping.to_s if object.ordering_users_count == 0
+    (object.shipping / object.ordering_users_count).to_s
+  end
+
   private
 
   def policy
     @policy ||= OrderPolicy.new(current_user, object)
   end
+
+  def current_user_dishes
+    object
```

Comment:
issue:
AFAIK we will do this query a few times.
WDYTA caching it with instance variable?

Reply:
You'll need to fetch those record first though...

Reply:
... and refactor some methods a little bit :smiley: